module/x/gravity/keeper/keeper.go

```diff
@@ -562,3 +562,65 @@ func (k Keeper) CreateContractCallTx(ctx sdk.Context, invalidationNonce uint64,
 	)
 	return newContractCallTx
 }
+
+/////////////////
+// MIGRATE     //
+/////////////////
+
+// Clean up all state associated a previous gravity contract and set a new contract. This is intended to run in the upgrade handler.
+func (k Keeper) MigrateGravityContract(ctx sdk.Context, newBridgeAddress string, bridgeDeploymentHeight uint64) {
+
+	// Delete Any Outgoing TXs.
+
+	prefixStoreOtx := prefix.NewStore(ctx.KVStore(k.storeKey), []byte{types.OutgoingTxKey})
```

Comment:
There is an `k.IterateOutgoingTxs` function too, any reason for not using that here?

Reply:
My read of of those iterator functions is that you can't actually delete a value from from the store in those iterators. Is that in correct?